### PR TITLE
Add docsite setup

### PR DESCRIPTION
**Pull request by nventuro**
Part of https://github.com/OpenZeppelin/docs.openzeppelin.com/issues/47

I'm trying out leaving the readme in .adoc here, which should make keeping the readme and index.adoc in sync much easier.

Following the convention I've been using, I've named the 'main' guide 'Getting Started', but I'm open to ideas on this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,210 +3,47 @@
...
-It requires two addresses with funds: one to act as the relayer, and one to act as its owner, who will register it on the hub. If these are not set when constructing the provider, they will default to the first two accounts on the local node.
-
-```js
-const { GSNDevProvider } = require("@openzeppelin/gsn-provider");
-const gsnDevProvider = new GSNDevProvider("http://localhost:8545", {
-  ownerAddress: accounts[0],
-  relayerAddress: accounts[1]
-});
-```
+* Check out docs/modules/ROOT/pages/getting-started.adoc[Getting Started] to use GSN Provider in a new project.
```
Comment (frangio) [inline README.md]:
```suggestion
* Check out link:docs/modules/ROOT/pages/getting-started.adoc[Getting Started] to use GSN Provider in a new project.
```

File: README.md [hunk 2]
```diff
@@ -3,210 +3,47 @@
...
+* The docs/modules/ROOT/pages/testing-gsn-applications.adoc[Testing GSN Applications] will teach you how to test your project locally.
```
Comment (frangio) [inline README.md]:
```suggestion
* The link:docs/modules/ROOT/pages/testing-gsn-applications.adoc[Testing GSN Applications] will teach you how to test your project locally.
```

**Comment by nventuro** [comment]
Yes, and I mistakenly pushed the updated branch to this repo instead of my fork, so they didn't show up here :man_facepalming:

File: package.json [hunk 3]
```diff
@@ -24,7 +24,7 @@
     }
   },
   "files": [
-    "README.md",
+    "README.adoc",
     "LICENSE",
```
Comment (frangio) [inline package.json]: These are implicit. See https://docs.npmjs.com/files/package.json#files:
> Certain files are always included, regardless of settings:
> - package.json
> - README
> - CHANGES / CHANGELOG / HISTORY
> - LICENSE / LICENCE
> - NOTICE
> - The file in the “main” field

Comment (nventuro) [reply package.json]: I suspected this, but didn't check. Thanks!

**Pull request merged**